### PR TITLE
Remaining structural changes to support new osb version

### PR DESCRIPTION
**Pull request by jayeshjk**
Changes in this PR:

- Adding support for `maintainance_info` , `instance_usable`, and `update_repeatable` features of new version in OSB Spec
- Adding config for supporting `metadata` field in binding response
- Adding support for specifying list of parameters returned for get instance endpoint

Feature: HCPCFS-2938

### CODE REVIEW

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-2 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-1 is succeeded.

**Review by anoopjb — APPROVED** [review]

File: interoperator/internal/properties/properties.go [hunk 1]
```diff
@@ -31,7 +33,7 @@ type Status struct {
 	Provision   InstanceStatus `yaml:"provision" json:"provision"`
 	Bind        GenericStatus  `yaml:"bind" json:"bind"`
 	Unbind      GenericStatus  `yaml:"unbind" json:"unbind"`
-	Deprovision GenericStatus  `yaml:"deprovision" json:"deprovision"`
+	Deprovision InstanceStatus `yaml:"deprovision" json:"deprovision"`
```
Comment (vivekzhere) [inline interoperator/internal/properties/properties.go:36]: Maybe as a different PR, we need to document the structure of the expected rendered status template

https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/docs/Interoperator-templates.md#status

**Review by vivekzhere — APPROVED** [review]
LGTM

File: interoperator/internal/properties/properties.go (hunk 1 above)
Comment (jayeshjk) [reply interoperator/internal/properties/properties.go:36]: As of now, empty values for new fields `instance_usable` and `update_repeatable` will be ignored. But I will modify the doc in next PR. Will list this item as pending in ticket.

**Pull request merged**